lib/job_simulation.rb

```diff
@@ -0,0 +1,56 @@
+class JobSimulation
+  attr_reader :working, :waiting
+
+  def initialize(jobs_available, job_seekers)
+    @working = CSFun::Stack.new
+    @waiting = CSFun::Queue.new
+
+    job_seekers.times do |i|
+      @waiting.enqueue("Worker \##{i + 1}")
+    end
+
+    while @working.size != jobs_available
+      @working.push( @waiting.dequeue )
+    end
+  end
+
+  def cycle(num=rand(1..6))
+    @roll = num
+    puts "Managers roll a #{@roll}"
+
+    @roll.times do
+      fired = @working.pop
+      puts "Fire #{fired}"
+      @waiting.enqueue(fired)
+    end
+
+    @roll.times do
+      hired = @waiting.dequeue
+      puts "Hire #{hired}"
+      @working.push(hired)
+    end
+  end
+end
+
+if __FILE__ == $0
```

Comment:
What's this? I don't know what ruby magic it is... explain so I don't have to google!